client/transaction-pool/src/testing/pool.rs

```diff
@@ -176,6 +179,10 @@ fn should_revalidate_during_maintenance() {
 	pool.api.push_block(1, vec![xt1.clone()]);
 
 	block_on(pool.maintain(&BlockId::number(1), &[]));
+
+	// maintaince is in background
+	block_on(futures_timer::Delay::new(std::time::Duration::from_millis(10)));
```

Comment:
This smells like it will become a flaky test.

Reply:
There is a fixed timeout just for tests, so it shouldn't be

Reply:
Is there a way to avoid relying on timeouts/intervals in tests? Like manually driving the process maybe?
If there is no way the duration should be dependent on the actual constant and not assume that we have 5 there so it should be 10 here.

Reply:
added const * 2 factor